### PR TITLE
Added authorization and limited routing for actions and displays

### PR DESCRIPTION
**Pull request by jeeheezy**

### CODE REVIEW

**Review by Gomezzzer — APPROVED** [review]
Looks good!

**Review by Gomezzzer — APPROVED** [review]
Looks good

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -4,16 +4,18 @@
...
-  </div>
-</div>
-
-<% @user.own_photos.each do |photo| %>
-  <div class="row mb-4">
+<% if !@user.private? || current_user.leaders.include?(@user) || current_user == @user %>
+  <div class="row mb-2">
     <div class="col-md-6 offset-md-3">
-      <%= render "photos/photo", photo: photo %>
+      <%= render "users/profile_nav", user: @user %>
```
Comment (wahnahbe) [inline app/views/users/show.html.erb:10]: ??

**Comment by wahnahbe** [comment]
lgtm

File: app/views/users/show.html.erb (hunk 1 above)
Comment (jeeheezy) [reply app/views/users/show.html.erb:10]: I too do not know why the diff is considering this when really it's just a difference on one line added on top

**Pull request merged**